mavis/extension_managers/RISCVExtensionManager.hpp

```diff
@@ -0,0 +1,570 @@
+            }
+
+            void enableBaseExtension(const std::string& ext)
+            {
+                if(base_extensions_.count(ext) == 0)
+                {
+                    throw InvalidBaseExtensionException(ext, base_extensions_);
+                }
+
+                enableExtension(ext);
+            }
+
+            void enableBaseExtension(const char ext)
+            {
+                enableBaseExtension(std::string(1, ext));
+            }
+
+            void setExtensionVersion(const std::string& ext, const uint32_t major_ver, const uint32_t minor_ver)
+            {
+                if(auto it = meta_extensions_.find(ext); it != meta_extensions_.end())
+                {
+                    for(const auto& child: it->second)
+                    {
+                        setExtensionVersion(child, major_ver, minor_ver);
+                    }
+                }
+                else if(auto alias_it = aliases_.find(ext); alias_it != aliases_.end())
+                {
+                    setExtensionVersion(alias_it->second, major_ver, minor_ver);
+                }
+                else if(!config_extensions_.count(ext))
+                {
+                    getExtensionInfo_(ext)->setVersion(major_ver, minor_ver);
+                }
+            }
+
+            void setExtensionVersion(const char ext, const uint32_t major_ver, const uint32_t minor_ver)
+            {
+                setExtensionVersion(std::string(1, ext), major_ver, minor_ver);
+            }
+    };
+
+    class RISCVExtensionManager : public ExtensionManager<RISCVExtensionInfo, RISCVXLENState>
+    {
+        private:
+            uint32_t xlen_ = 0;
+
+            static inline std::string toLowercase_(const std::string& str)
+            {
+                std::string lowercase;
+                lowercase.reserve(str.size());
+                std::transform(str.begin(), str.end(), std::back_inserter(lowercase), [](const char chr) { return std::tolower(chr); });
+                return lowercase;
+            }
+
+            static bool getBoolJSONValue_(const nlohmann::json& jobj, const std::string& key)
+            {
+                if(const auto it = jobj.find(key); it != jobj.end())
+                {
+                    return *it;
+                }
+
+                return false;
+            }
+
+            template<typename ValType, typename OtherValType>
+            static bool isOneOf_(const ValType val, const OtherValType other)
+            {
+                return (val == other);
+            }
+
+            template<typename ValType, typename OtherValType, typename ... OtherValTypes>
+            static bool isOneOf_(const ValType val, const OtherValType other, const OtherValTypes... rest)
+            {
+                return (val == other) || isOneOf_(val, rest...);
+            }
+
+            static uint32_t digitToInt_(const char digit)
+            {
+                return digit - '0';
```

Comment:
Consider adding this assert: 
assert(std::isdigit(static_cast<unsigned char>(ch)) && "Character is not a digit");

Or use std::atoi(&digit)

Reply:
I added the assert. `std::atoi(&digit)` wouldn't work because there's no guarantee `&digit` would be null-terminated.